Azure-Samples/Cognitive-Services-Voice-Assistant
# Add kws performance feature to UWP VA sample

## Purpose
* Added an optional key for enabling keyword performance logging - this option is false by default, to enable set EnableKwsLogging: true in config.json
* Added an optional key for enabling hardware kws - this option is false by default, to enable set EnableHardwareDetector: true and if the hardware keyword is on the device then set SetModelData: false
* If EnableKwsLogging is true a KwsPerformanceMetrics.csv file will be created in the LocalState Folder
* Keyword Activation and Confirmation models can be specified using a file path
* Added an optional key for setting property for the DialogServiceConnector object.
   - To turn on KWS+KWV+SR only, enter the following key value pair in the SetProperty object
   - ``` "SPEECH-Endpoint": "wss://westus.convai.speech.microsoft.com/orchestrate/api/v1" ```

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:

## How to Test
*  Launch the app
* Open the config file and add "EnableKwsLogging: true"
* Apply config and voice activate the app
* Click open logs and you should see the KwsPerformanceMetrics.csv file
